src/reputation.c

```diff
@@ -473,53 +535,31 @@ int SRepLoadFileFromFD(SRepCIDRTree *cidr_ctx, FILE *fp)
                 PrintInet(AF_INET6, (const void *)&a.address, ipstr, sizeof(ipstr));
                 SCLogDebug("%s %u %u", ipstr, cat, value);
             }
+            SRepLoadLine(&a, cat, value);
+        }
+    }
 
-            Host *h = HostGetHostFromHash(&a);
-            if (h == NULL) {
-                SCLogError(SC_ERR_NO_REPUTATION, "failed to get a host, increase host.memcap");
-                break;
-            } else {
-                //SCLogInfo("host %p", h);
-
-                if (h->iprep == NULL) {
-                    h->iprep = SCMalloc(sizeof(SReputation));
-                    if (h->iprep != NULL) {
-                        memset(h->iprep, 0x00, sizeof(SReputation));
-
-                        HostIncrUsecnt(h);
-                    }
-                }
-                if (h->iprep != NULL) {
-                    SReputation *rep = h->iprep;
-
-                    /* if version is outdated, it's an older entry that we'll
-                     * now replace. */
-                    if (rep->version != SRepGetVersion()) {
-                        memset(rep, 0x00, sizeof(SReputation));
-                    }
-
-                    rep->version = SRepGetVersion();
-                    rep->rep[cat] = value;
-
-                    SCLogDebug("host %p iprep %p setting cat %u to value %u",
-                        h, h->iprep, cat, value);
-#ifdef DEBUG
-                    if (SCLogDebugEnabled()) {
-                        int i;
-                        for (i = 0; i < SREP_MAX_CATS; i++) {
-                            if (rep->rep[i] == 0)
-                                continue;
-
-                            SCLogDebug("--> host %p iprep %p cat %d to value %u",
-                                    h, h->iprep, i, rep->rep[i]);
-                        }
-                    }
-#endif
-                }
+    return 0;
+}
 
-                HostRelease(h);
-            }
+int SRepLoadLineFromUnix(SRepCIDRTree *cidr_ctx,
+                         char *ipaddress,
+                         int cat, int val)
+{
+    if (strchr(ipaddress, '/') != NULL) {
+        int r = SRepCIDRAddNetblock(cidr_ctx, ipaddress, cat, val);
```

Comment:
How does this affect thread safety?

Reply:
that would mean every lookup is affected by this lock too, so each detect thread in some cases for each packet (normally per flow, but for packets w/o flow it's per packet)